test/document_test.rb

```diff
@@ -443,6 +443,17 @@ def setup
     assert_equal [], second_doc.foo
   end
 
+  test 'attachments work properly' do
```

Comment:
In all honesty though, the VCR gem is not so bad at this kind of stuff, because you can periodically update the cassette.

Basically you run your test suite with a direct connection to couchdb, VCR records the request/response data into a yaml file, and then from that point the connection is severed and it uses the yaml file to 'replay' the requests.